python/semantic_kernel/connectors/ai/onnx/services/onnx_gen_ai_text_completion.py

```diff
@@ -0,0 +1,128 @@
+            env_file_path (str | None): Use the environment settings file as a fallback
+                to environment variables.
+            env_file_encoding (str | None): The encoding of the environment settings file.
+        """
+        try:
+            settings = OnnxGenAISettings.create(
+                folder=ai_model_path,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as e:
+            raise ServiceInitializationError(f"Invalid settings for OnnxGenAITextCompletion: {e}")
+
+        if ai_model_id is None:
+            ai_model_id = settings.folder
+
+        super().__init__(
+            ai_model_id=ai_model_id,
+            ai_model_path=settings.folder,
+        )
+
+    @override
+    async def _inner_get_text_contents(
+        self,
+        prompt: str,
+        settings: PromptExecutionSettings,
+    ) -> list[TextContent]:
+        """This is the method that is called from the kernel to get a response from a text-optimized LLM.
+
+        Args:
+            prompt (str): The prompt to send to the LLM.
+            settings (OnnxGenAIPromptExecutionSettings): Settings for the request.
+
+        Returns:
+            List[TextContent]: A list of TextContent objects representing the response(s) from the LLM.
+        """
+        if not isinstance(settings, OnnxGenAIPromptExecutionSettings):
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, OnnxGenAIPromptExecutionSettings)  # nosec
+
+        new_tokens = ""
+        async for new_token in self._generate_next_token(prompt, settings):
+            new_tokens += new_token
+
+        return [
+            TextContent(
+                text=new_tokens,
+                ai_model_id=self.ai_model_id,
+            )
+        ]
+
+    @override
+    async def _inner_get_streaming_text_contents(
+        self,
+        prompt: str,
+        settings: PromptExecutionSettings,
+    ) -> AsyncGenerator[list[StreamingTextContent], Any]:
+        """Streams a text completion using a Onnx model.
+
+        Note that this method does not support multiple responses.
+
+        Args:
+            prompt (str): Prompt to complete.
+            settings (OnnxGenAIPromptExecutionSettings): Request settings.
+
+        Yields:
+            List[StreamingTextContent]: List of StreamingTextContent objects.
+        """
+        if not isinstance(settings, OnnxGenAIPromptExecutionSettings):
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, OnnxGenAIPromptExecutionSettings)  # nosec
+
+        async for new_token in self._generate_next_token(prompt, settings):
+            yield [
+                StreamingTextContent(
+                    choice_index=0, inner_content=new_token, text=new_token, ai_model_id=self.ai_model_id
```

Comment:
We don't get any choice index info from the `generate_next_token` method?

Reply:
Onnx works on a very low level, the reason for choices is the Autogenerative AI Loop around Models. The Model produces a probability distribution for each token. 

Then Search Algorithm will determine the next tokens, most popular is the Greedy Search, it just takes the token with the highest probablity. If we do a greedy search with Onnx we will always get one choice.

But there are also other Search Algorithm like the Beam Search, you can specifiy it with `num_beans=n`, then the model will come up with `n` the user can pick from.

I also implemented the Onnx Connector now to work with multiple choices when BeamSearch is activated 👍